core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -109,7 +109,16 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def createNullValue(cls: Class[_]): AnyRef = {
     if (cls.isPrimitive) {
-      new java.lang.Byte(0: Byte) // Should be convertible to any primitive type
+      if (cls == java.lang.Boolean.TYPE) {
```

Comment:
The following code will fail:
```Scala
case class X(x: Char)

object Foo1 extends App {
  val c = classOf[X]
  val x = c.getConstructors()(0)
  println(x)
  println(x.newInstance(new java.lang.Byte(0: Byte)))
}
```
I don't know why `char` is also a special case.